operator/pkg/helmreconciler/reconciler.go

```diff
@@ -242,6 +243,18 @@ func (h *HelmReconciler) CheckSSAEnabled() bool {
 		// There is a mutatingwebhook in gke that would corrupt the managedFields, which is fixed in k8s 1.18.
 		// See: https://github.com/kubernetes/kubernetes/issues/96351
 		if k8sVer >= 18 {
+			// todo(kebe7jun) a more general test method
+			// API Server does not support detecting whether ServerSideApply is enabled
+			// through the API for the time being.
+			ns, err := h.clientSet.CoreV1().Namespaces().Get(context.TODO(), constants.KubeSystemNamespace, v12.GetOptions{})
+			if err != nil {
+				scope.Warnf("failed to get namespace: %v", err)
+				return false
+			}
+			if ns.ManagedFields == nil {
```

Comment:
Is this a reliable mechanism?

Reply:
Currently Kubernetes doesn't provide an interface for clients to determine SSA enablement status, so we'll have to go by that result for now, or do you have a better way?